### PR TITLE
Add new editable fields in the LOADS tab in the spreadsheet : name, type, p0, q0.

### PR DESCRIPTION
**Pull request by dbraquart**
We also want to do it using the 'Update Load' existing hypothesis rather than a generic groovy update.

### CODE REVIEW

File: src/components/network/config-tables.js [hunk 1]
```diff
@@ -711,12 +729,26 @@ export const TABLES_DEFINITIONS = {
                 dataKey: 'p0',
                 numeric: true,
                 fractionDigits: 1,
+                changeCmd: 'equipment.setP0({})\n',
+                editor: ({ equipment, ...props }) =>
```
Comment (archendel) [inline src/components/network/config-tables.js:733]: undefined props don't need to be specified, I think you can simplify this to :
editor: NumericalField,

File: src/components/network/config-tables.js [hunk 2]
```diff
@@ -674,10 +681,21 @@ export const TABLES_DEFINITIONS = {
                 id: 'Name',
                 dataKey: 'name',
                 columnWidth: MEDIUM_COLUMN_WIDTH,
+                changeCmd: "equipment.setName('{}')\n",
+                editor: ({ equipment, ...props }) =>
```
Comment (archendel) [inline src/components/network/config-tables.js:685]: I think you can simplify to :
editor: NameField,

File: src/components/network/config-tables.js [hunk 3]
```diff
@@ -711,12 +729,26 @@ export const TABLES_DEFINITIONS = {
...
+                        ...props,
+                    }),
             },
             {
                 id: 'ConstantReactivePower',
                 dataKey: 'q0',
                 numeric: true,
                 fractionDigits: 1,
+                changeCmd: 'equipment.setQ0({})\n',
+                editor: ({ equipment, ...props }) =>
```
Comment (archendel) [inline src/components/network/config-tables.js:746]: same as above

File: src/components/network/equipment-table-editors.js [hunk 4]
```diff
@@ -86,8 +89,69 @@ export const NumericalField = ({
...
+        <TextField
+            defaultValue={defaultValue}
+            onChange={(ev) => setter(ev.target.value)}
+            {...props}
+            size={'small'}
+            margin={'none'}
+            style={{ ...style, padding: 0 }}
+            inputProps={{
+                style: {
+                    textAlign: 'center',
```
Comment (archendel) [inline src/components/network/equipment-table-editors.js:124]: I think it looks better without centering the text, it's my opinion though

File: src/components/network/equipment-table-editors.js [hunk 5]
```diff
@@ -86,8 +89,69 @@ export const NumericalField = ({
...
+                    fontSize: 'small',
+                    flexGrow: 1,
```
Comment (archendel) [inline src/components/network/equipment-table-editors.js:126]: what is the flexGrow useful for ?

File: src/components/network/equipment-table-editors.js [hunk 6]
```diff
@@ -86,8 +89,69 @@ export const NumericalField = ({
...
+};
+
+export const EnumField = ({ enumList, setter, defaultValue, ...props }) => {
+    return (
+        <Select
+            defaultValue={defaultValue || ''}
+            onChange={(ev) => setter(ev.target.value)}
+            size={'medium'}
+            margin={'none'}
+            MenuProps={{
```
Comment (archendel) [inline src/components/network/equipment-table-editors.js:140]: why is it needed ?

**Review by archendel — CHANGES_REQUESTED** [review]
Test : OK
Code review : some small changes

File: src/components/network/equipment-table-editors.js (hunk 5 above)
Comment (dbraquart) [reply src/components/network/equipment-table-editors.js:143]: it is useless. I copy/paste all the style from another TextField Editor to look like the same. I will remove it (in both)

File: src/components/network/equipment-table-editors.js (hunk 6 above)
Comment (dbraquart) [reply src/components/network/equipment-table-editors.js:157]: I dont know. I copy/paste this part from another Select Editor to look like the same. To be verified

File: src/components/network/config-tables.js (hunk 2 above)
Comment (dbraquart) [reply src/components/network/config-tables.js:685]: yes, right. I will simplify as requested

File: src/components/network/config-tables.js (hunk 3 above)
Comment (dbraquart) [reply src/components/network/config-tables.js:746]: yes, right. I will simplify as requested

File: src/components/network/config-tables.js (hunk 1 above)
Comment (dbraquart) [reply src/components/network/config-tables.js:733]: yes, right. I will simplify as requested

File: src/components/network/equipment-table-editors.js (hunk 6 above)
Comment (dbraquart) [reply src/components/network/equipment-table-editors.js:157]: I got the answer: it was to limit the height of the select one opened. I dont need it in my "enum" case, so I'll remove it.

**Review by archendel — APPROVED** [review]
One small visual trouble, the dropdown list field seems to be slightly bigger than the textfields

File: src/components/network/equipment-table-editors.js [hunk 7]
```diff
@@ -82,12 +90,76 @@ export const NumericalField = ({
...
+                    step: 'any',
                 }}
             />
-        </Tooltip>
+        );
+    }
+
+    return (
+        <div>
+            {min !== undefined || max !== undefined ? (
```
Comment (EstherDarkish) [inline src/components/network/equipment-table-editors.js:127]: If there is no "min" values, there should not be "min: " in the tooltip.

File: src/components/network/equipment-table-editors.js [hunk 8]
```diff
@@ -82,12 +90,76 @@ export const NumericalField = ({
...
+    );
+};
+
+export const NameField = ({ defaultValue, setter, style, ...props }) => {
+    const [error, setError] = useState(false);
+
+    const validateChange = useCallback(
+        (ev) => {
+            const newVal = ev.target.value;
+            setError(newVal.length === 0);
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/components/network/equipment-table-editors.js]: yes in deed. cf previous commit ffa9e9e

File: src/components/network/equipment-table-editors.js (hunk 7 above)
Comment (dbraquart) [reply src/components/network/equipment-table-editors.js:127]: agreed. to be changed

Comment (EstherDarkish) [reply src/components/network/equipment-table-editors.js]: Maybe change the new code to not have a lot of ternary operators ?

Comment (dbraquart) [reply src/components/network/equipment-table-editors.js]: agreed. to be changed in next commit

**Pull request merged**